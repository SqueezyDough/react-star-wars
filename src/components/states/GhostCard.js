import React from 'react'
import styled from 'styled-components'

function GhostCard ({ className }) {
  return (
    <div className={className}>
      <article className='card'>
        <span className='card__short-title'>L</span>
        <h1 className='card__title'>Loading...</h1>
      </article>
    </div>
  )
}

export default styled(GhostCard)` 
  background-color: ${({ theme }) => theme.primaryColorLight};
  transition: background ${({ theme }) => theme.transition};
  text-align: center;
  color:  ${({ theme }) => theme.secondaryColorDark};
  animation: pulse 1s infinite alternate;

  @keyframes pulse {
    0% {
      opacity: .2;
    }
  }

  &:first-of-type {
    grid-column: 2;
  }

  .card {
    padding: 10%;
  }

  .card__short-title {
    font-family: ${({ theme }) => theme.fontTitles}
    font-size: ${({ theme }) => theme.fontSizeXXL};
    font-weight: bold;
  }
`
