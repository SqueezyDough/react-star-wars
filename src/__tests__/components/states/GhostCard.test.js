import React from 'react'
import renderer from 'react-test-renderer'
import GhostCard from '../../../components/states/GhostCard'

it('renders correctly', () => {
  const tree = renderer
    .create(<GhostCard />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
