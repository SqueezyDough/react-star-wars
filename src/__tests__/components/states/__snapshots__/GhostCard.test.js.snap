// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  -webkit-transition: background;
  transition: background;
  text-align: center;
  -webkit-animation: pulse 1s infinite alternate;
  animation: pulse 1s infinite alternate;
}

.c0:first-of-type {
  grid-column: 2;
}

.c0 .card {
  padding: 10%;
}

.c0 .card__short-title {
  font-weight: bold;
}

<div
  className="c0"
>
  <article
    className="card"
  >
    <span
      className="card__short-title"
    >
      L
    </span>
    <h1
      className="card__title"
    >
      Loading...
    </h1>
  </article>
</div>
`;
